src/channels/channel.ts

```diff
@@ -136,4 +164,32 @@ export class Channel {
             Log.info(`[${new Date().toLocaleTimeString()}] - ${socket.id} joined channel: ${channel}`);
         }
     }
+
+    /**
+     * Check if client is a client event
+     *
+     * @param  {string} event
+     * @return {boolean}
+     */
+    isClientEvent(event: string): boolean {
+        let isClientEvent = false;
+
+        this._clientEvents.forEach(clientEvent => {
+            let regex = new RegExp(clientEvent.replace('\*', '.*'));
+            if (regex.test(event)) isClientEvent = true;
+        });
+
+        return isClientEvent;
+    }
+
+    /**
+     * Check if a socket has joined a channel.
+     *
+     * @param socket
+     * @param channel
+     * @returns {boolean}
+     */
+    isInChannel(socket: any, channel: string): boolean {
+        return this.io.sockets.adapter.sids[socket.id][channel];
+    }
```

Comment:
https://github.com/socketio/socket.io/blob/master/docs/API.md#socketrooms
`socket.rooms`: 
```js
{ Ny8TtuSKjpoSOKehAAAA: 'Ny8TtuSKjpoSOKehAAAA',
  'presence-chat.14': 'presence-chat.14',
  'private-App.Park.14': 'private-App.Park.14' }
```
`this.io.sockets.adapter.sids[socket.id]`: 
```js
{ Ny8TtuSKjpoSOKehAAAA: true,
  'presence-chat.14': true,
  'private-App.Park.14': true }
```

So I guess both should work.